test/credentials_test.dart

```diff
@@ -226,4 +226,119 @@ Future<void> main([List<String>? args]) async {
+  // Run first two steps (test 1 and test 2) to create and confirm the user.
+  // Before running test 2 set the variables with token details received as link query parameters in the confirmation email.
+  // Then run test 3, then make sure you have recieved an email with link for reset password.
+  // Copy token and tokenId from the email link query parameters.
+  // Set the variables with token details and then run test 4.
+  // Test 4 will set the pasword and will login the user with the new password.
+  group("Manual test: Email/Password - reset password", () {
+    // Enter a valid email that is not registered
+    const String validUsername = "valid_email@mail.com";
+
+    baasTest('Manual test 1 (resetPassword). Register a valid user', (configuration) async {
+      final app = App(configuration);
+      final authProvider = EmailPasswordAuthProvider(app);
+      await authProvider.registerUser(validUsername, strongPassword);
+    }, appName: "emailConfirm", skip: "It is a manual test");
+
+    baasTest('Manual test 2 (resetPassword). Take recieved token from the received email and confirm the user', (configuration) async {
+      // Enter valid token and tokenId from the received email
+      String token = "3e23e2e689fe1fdbbb51d3c090a216a4195a4f0a004a578787618d9dd39c791f4169511ee3820e4b6fa2bfdc14430f1e58356223d78e3bed3c86042c3a91a4db";
+      String tokenId = "6278cecd9106aa5b645999ba";
+
+      final app = App(configuration);
+      final authProvider = EmailPasswordAuthProvider(app);
+      await authProvider.confirmUser(token, tokenId);
+      final user = await loginWithRetry(app, Credentials.emailPassword(validUsername, strongPassword));
+      expect(user, isNotNull);
+    }, appName: "emailConfirm", skip: "It is a manual test");
+
+    baasTest('Manual test 3 (resetPassword). Reset user password email', (configuration) async {
+      final app = App(configuration);
+      final authProvider = EmailPasswordAuthProvider(app);
+      await authProvider.resetPassword(validUsername);
+    }, appName: "emailConfirm", skip: "It is a manual test");
+
+    baasTest('Manual test 4 (resetPassword). Take recieved token from the email and complete resetting new password', (configuration) async {
+      // Make sure you have recieved an emails with hyperlink ResetPassword.
+      // Find the token and tokenId in the query parameters of the link received in the email and enter them in the following variables.
+      String token = "fb485146b15497209a9d1b67128ae29199cdff2c26f389e0ee5d52ae6bc4228e5738b29256eade976e24767804bfb4dc68198075e3a461cd4f73864901fb09be";
+      String tokenId = "6272619334f5b3a770a7dc2c";
+
+      final app = App(configuration);
+      final authProvider = EmailPasswordAuthProvider(app);
+      String newPassword = "RWE@#EDE";
+      await authProvider.completeResetPassword(newPassword, token, tokenId);
+      final user = await loginWithRetry(app, Credentials.emailPassword(validUsername, strongPassword));
```

Comment:
Does this require `loginWithRetry`? My expectation is that the password change propagates immediately.

Reply:
I'm not sure. Where can I check it?

Reply:
I don't think we have docs/way to check that - my suggestion would be to just use `app.logIn` directly rather than the version that retries.

Reply:
Is it only confirmation process that takes more time to be able to login?

Reply:
In my experience, it's confirmation when autoconfirm is set to `true` or when using a confirmation function. I guess the reason there is that there's no explicit "confirm" call that you invoke, so the `register` call does not wait for confirmation to run. Whenever you have a "completeXXX" method, that typically completes when the change has been persisted and propagated.